qr-code/src/main/java/com/adyen/checkout/qrcode/QRCodeComponent.kt

```diff
@@ -81,6 +87,12 @@ class QRCodeComponent(application: Application, configuration: QRCodeConfigurati
     @Throws(ComponentException::class)
     override fun handleActionInternal(activity: Activity, action: Action) {
         if (action !is QrCodeAction) throw ComponentException("Unsupported action")
+        if (!PROVIDER.requiresView(action)) {
+            Logger.d(TAG, "Action does not require a view, redirecting.")
+            isPollingEnabled = false
+            redirectDelegate.makeRedirect(activity, action.url)
+            return
+        }
         paymentMethodType = action.paymentMethodType
         qrCodeData = action.qrCodeData
```

Comment:
Could the `qrCodeData` be the URL in case of a redirect? Is there any scenario where there are both a qrCode and a URL?

Reply:
Yeah it could if the backend sends it like that, but it shouldn't. We will just try to redirect any QR code action that is not within the supported payment methods (only PIX now) and has a URL

Reply:
What I meant was for us to set like `qrCodeData = action.url` in the case of a redirect.
Like, this came up because I was thinking of how different the behavior of a regular QrCode and a Redirect should be here. For example, maybe we should also populate `paymentMethodType` in the redirect flow?

This current implementation is just returning and it makes me feel like it's sort of an inconsistent state. Does this make sense?